tests/asahirov/test_homework_6.py

```diff
@@ -0,0 +1,154 @@
+from operator import add, mul
+
+
+def min(*args, key=None):
+    """
+    Return min element from scope of value
+    @param args: values for min value retrieve
+    @param key: if abs retrieve by absolute value
+    @returns min_element: Minimum value from scope
+    """
+    if key is None:
+        key = lambda x: x
+
+    min_value = None
+    min_element = None
+    for element in args:
+        value = key(element)
+        if min_value is None or value < min_value:
+            min_value = value
+            min_element = element
+
+    return min_element
+
+
+def max(*args, key=None):
+    """
+    Return max element from scope of value
+    @param args: values for max value retrieve
+    @param key: if abs retrieve by absolute value
+    @returns max_element: Maximum value from scope
+    """
+    if key is None:
+        key = lambda x: x
+
+    max_value = None
+    max_element = None
+    for element in args:
+        value = key(element)
+        if max_value is None or value > max_value:
+            max_value = value
+            max_element = element
+
+    return max_element
+
+
+def sorted(*args, key=None, reverse=False):
+    """
+    Return sorted elements list
+    @param args: values for sort
+    @param key: if abs sort by absolute value
+    @param reverse: if True return reversed result
+    @returns sorted_list: sorted list of values
+    """
+    items_count = len(args)
+    elements = list(args)
+    if key is None:
+        key = lambda x: x
+    sorted_list = []
+    for i in range(items_count):
+        if reverse:
+            value = max(*elements, key=key)
+            sorted_list.append(value)
+            elements.remove(value)
+        else:
+            value = min(*elements, key=key)
+            sorted_list.append(value)
+            elements.remove(value)
```

Comment:
you can avoid this code duplication if first select function - either min or max
and then use function to calculate value